Crash: Adding an order note causes the app to crash

### Expected behavior
Adding an order note routes back to the order detail view. 

### Actual behavior
In production: the app crashes:
```
E  FATAL EXCEPTION: main
E  Process: com.woocommerce.android, PID: 7888
E  java.lang.RuntimeException: Unable to start activity ComponentInfo{com.woocommerce.android/com.woocommerce.android.ui.main.MainActivity}: java.lang.NullPointerException: Attempt to invoke virt
   ual method 'android.view.View android.widget.FrameLayout.getChildAt(int)' on a null object reference
E      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2914)
E      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3049)
E      at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
E      at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
E      at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
E      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1809)
E      at android.os.Handler.dispatchMessage(Handler.java:106)
E      at android.os.Looper.loop(Looper.java:193)
E      at android.app.ActivityThread.main(ActivityThread.java:6680)
E      at java.lang.reflect.Method.invoke(Native Method)
E      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
E      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
E  Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.View android.widget.FrameLayout.getChildAt(int)' on a null object reference
E      at com.woocommerce.android.ui.base.TopLevelFragment.updateParentViewState(TopLevelFragment.kt:136)
E      at com.woocommerce.android.ui.base.TopLevelFragment.onBackStackChanged(TopLevelFragment.kt:120)
E      at com.woocommerce.android.ui.orders.OrderListFragment.onBackStackChanged(OrderListFragment.kt:267)
E      at android.support.v4.app.FragmentManagerImpl.reportBackStackChanged(FragmentManager.java:2704)
E      at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2411)
E      at android.support.v4.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2338)
E      at android.support.v4.app.FragmentManagerImpl.popBackStackImmediate(FragmentManager.java:854)
E      at android.support.v4.app.FragmentManagerImpl.popBackStackImmediate(FragmentManager.java:808)
E      at com.woocommerce.android.ui.main.MainActivity.clearFragmentBackStack(MainActivity.kt:448)
E      at com.woocommerce.android.ui.main.MainActivity.switchFragment(MainActivity.kt:387)
E      at com.woocommerce.android.ui.main.MainActivity.switchFragment$default(MainActivity.kt:383)
E      at com.woocommerce.android.ui.main.MainActivity.initFragment(MainActivity.kt:358)
E      at com.woocommerce.android.ui.main.MainActivity.onCreate(MainActivity.kt:121)
E      at android.app.Activity.performCreate(Activity.java:7144)
E      at android.app.Activity.performCreate(Activity.java:7135)
E      at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
E      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2894)
```

With my latest changes to redesign the parent fragments, the app does not crash, but upon submitting the note, the user is taken to the order list view instead of order detail and the note is not saved.

### Steps to reproduce the behavior
1. Open an order into order detail
2. Click **Add a note**
3. Type up a message and click **ADD**
-- app crashes

##### Tested on [device], Android [version], WCAndroid [version]
Physical device: Pixel 2 running Android 9

--- a/RELEASE-NOTES.txt
+++ b/RELEASE-NOTES.txt
@@ -1,0 +1,2 @@
+Bugfixes
+- Issue with order notes not saving properly during activity restore (low memory situations) resolved

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt
@@ -153,7 +153,7 @@ class MainActivity : AppCompatActivity(),
         // Restore the current navigation position
         savedInstanceState?.also {
             val id = it.getInt(STATE_KEY_POSITION, BottomNavigationPosition.DASHBOARD.id)
-            bottomNavView.currentPosition = findNavigationPositionById(id)
+            bottomNavView.restoreSelectedItemState(id)
         }
     }
 

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainNavigationView.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainNavigationView.kt
@@ -69,6 +69,16 @@ class MainNavigationView @JvmOverloads constructor(
         updateCurrentPosition(navPos, true)
     }
 
+    /**
+     * For use when restoring the navigation bar after the host activity
+     * state has been restored.
+     */
+    fun restoreSelectedItemState(itemId: Int) {
+        assignNavigationListeners(false)
+        selectedItemId = itemId
+        assignNavigationListeners(true)
+    }
+
     fun showNotificationBadge(show: Boolean) {
         with(badgeView) {
             if (show && visibility != View.VISIBLE) {

